Sources/Sentry/SentrySwizzle.m

```diff
@@ -102,11 +104,59 @@ @implementation SentrySwizzle
 
     pthread_mutex_lock(&gLock);
 
+#if TEST
+    IMP currImp = class_getMethodImplementation(classToSwizzle, selector);
+    [GULSwizzlingCache cacheCurrentIMP:currImp
+                             forNewIMP:newIMP
+                              forClass:classToSwizzle
+                          withSelector:selector];
+#endif
+
     originalIMP = class_replaceMethod(classToSwizzle, selector, newIMP, methodType);
 
     pthread_mutex_unlock(&gLock);
 }
 
+static void
+unswizzle(Class classToSwizzle, SEL selector)
+{
+#if TEST
+    static pthread_mutex_t gLock = PTHREAD_MUTEX_INITIALIZER;
+    pthread_mutex_lock(&gLock);
+
+    NSCAssert(
+        classToSwizzle != nil && selector != nil, @"You cannot unswizzle a nil class or selector.");
+    Method method = class_getInstanceMethod(classToSwizzle, selector);
+
+    NSCAssert(method, @"Couldn't find the method you're unswizzling in the runtime.");
+    IMP originalImp = [[GULSwizzlingCache sharedInstance] cachedIMPForClass:classToSwizzle
+                                                               withSelector:selector];
+    NSCAssert(originalImp, @"This class/selector combination hasn't been swizzled");
+    IMP currentImp = method_setImplementation(method, originalImp);
+    __unused BOOL didRemoveBlock = imp_removeBlock(currentImp);
+    NSCAssert(didRemoveBlock, @"Wasn't able to remove the block of a swizzled IMP.");
+    [[GULSwizzlingCache sharedInstance] clearCacheForSwizzledIMP:currentImp
+                                                        selector:selector
+                                                          aClass:classToSwizzle];
+
+    pthread_mutex_unlock(&gLock);
+
+#endif
+}
+
++ (void)resetSwizzling
+{
+    NSArray *cached = [[GULSwizzlingCache sharedInstance] cachedClasses];
+    for (int i = 0; i < cached.count; i++) {
+        CFArrayRef item = (__bridge CFArrayRef)cached[i];
+        Class class = (Class)CFArrayGetValueAtIndex(item, 0);
+        SEL selector = (SEL)CFArrayGetValueAtIndex(item, 1);
+        unswizzle(class, selector);
+    }
+
+    [swizzledClassesDictionary() removeAllObjects];
```

Comment:
`h`: swizzledClassesDictionary is not thread safe and all the other usages use a `@synchronized` block.
```suggestion
    @synchronized(swizzledClassesDictionary()) {
        [swizzledClassesDictionary() removeAllObjects];
    }
```